app/ports/views.py

```diff
@@ -246,12 +246,11 @@ def all_builds_filter(request):
 
 def stats(request):
     days = request.GET.get('days', 30)
-    allowed_days = [7, 30, 90, 180, 365]
-    try:
-        if int(days) not in allowed_days:
-            return HttpResponse("'days' received an invalid value.")
-    except ValueError:
-            return HttpResponse("'days' must be an integer.")
+
+    # Validate days
+    check, message = validate_stats_days(days)
+    if check is False:
+        return HttpResponse(message)
```

Comment:
One option would be to simply round to the nearest "valid" value instead, and maybe drop some invalid arguments instead of showing the error, but let's not touch this now..